src/L2Factory.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.20;
+
+import {L2Deployer} from "./L2Deployer.sol";
+import {L2Token} from "@zkevm-stb/L2Token.sol";
+import {L2Escrow} from "@zkevm-stb/L2Escrow.sol";
+import {L2TokenConverter} from "@zkevm-stb/L2TokenConverter.sol";
+
+/// @title Factory for deployment of L2 Contracts.
+contract L2Factory {
```

Comment:
How do we make this contract deployed to the same address on all chains?

Reply:
This is the one that would be done through Nicks Method

Requires that the L1 Deployer and bridge address are the same though